File: apps/api/app/schemas/notification.py
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.schemas.common import PaginationMeta


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    type: str
    title: str
    body: Optional[str] = None
    source_type: Optional[str] = None
    source_id: Optional[str] = None
    project_id: Optional[str] = None
    metadata: Optional[dict] = None
    read: bool
    read_at: Optional[str] = None
    created_at: Optional[str] = None


class NotificationListResponse(BaseModel):
    data: list[NotificationResponse]
    meta: PaginationMeta


class UnreadCountResponse(BaseModel):
    count: int


class NotificationPreferencesUpdate(BaseModel):
    email_enabled: Optional[bool] = None
    email_categories: Optional[dict] = None

File: apps/web/src/types/notification.ts
export interface Notification {
  id: string;
  type: string;
  title: string;
  body?: string | null;
  source_type?: string | null;
  source_id?: string | null;
  project_id?: string | null;
  metadata?: Record<string, unknown>;
  read: boolean;
  read_at?: string | null;
  created_at: string;
}

export interface NotificationPreferences {
  email_enabled: boolean;
  email_categories: {
    assigned_to_me: boolean;
    status_changes: boolean;
    mentions: boolean;
    approaching_deadlines: boolean;
    bid_invitations: boolean;
    pay_app_decisions: boolean;
    meeting_scheduled: boolean;
    meeting_minutes: boolean;
    daily_summary: boolean;
  };
}
